src/client/pages/Offer/ProductSelector/Selector.tsx

```diff
@@ -22,6 +22,57 @@ export type SelectorProps = {
   additionalProducts: Array<Product>
 }
 
+type State = {
+  standaloneProducts: Record<string, boolean>
+  additionalProducts: Record<string, boolean>
+}
+
+const getStateFromURL = ({
+  standaloneProducts,
+  additionalProducts,
+  searchParams,
+}: {
+  standaloneProducts: Array<Product>
+  additionalProducts: Array<Product>
+  searchParams: string
+}): State => {
+  const selectedInsuranceTypes = new Set(
+    new URLSearchParams(searchParams).getAll(SEARCH_PARAM),
+  )
+
+  const getProductIdByValueMap = (quotes: Array<Product>) =>
```

Comment:
Nit: Constructing with `Object.fromEntries()` might be easier to read than manual reduce.  And it's more efficient since you don't copy on every iteration